### PR TITLE
Auditing for Vote, VotePost, VoteOption and User

### PR DESCRIPTION
**Pull request by navro**

### CODE REVIEW

File: app/helpers/votes_helper.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+module VotesHelper
+  def split_audit_hashes(input, model)
+    values = []
+    eval(input).each do |key, value|
```
Comment (davidwessman) [inline app/helpers/votes_helper.rb]: What does eval do in this case?

Comment (navro) [inline app/helpers/votes_helper.rb]: > What does eval do in this case?

It converts the string from audit to a hash. Is there a safer way?

Comment (davidwessman) [inline app/helpers/votes_helper.rb]: I think we should use a JSON-store in postgres instead.
http://blog.arkency.com/2014/10/how-to-persist-hashes-in-rails-applications-with-postgresql/
Scroll down to JSON.

File: app/models/audit.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+ class Audit < ActiveRecord::Base
+  belongs_to :auditable, polymorphic: true
+  belongs_to :user, -> { with_deleted }
+  belongs_to :updater, class_name: 'User'
+  belongs_to :vote, -> { with_deleted }
+
+  validates :auditable, presence: true
+end
```
Comment (navro) [inline app/models/audit.rb]: I disagree!

Comment (davidwessman) [inline app/models/audit.rb]: The space is gone? ;)

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -4,6 +4,9 @@ class ApplicationController < ActionController::Base
   before_action :configure_permitted_devise_parameters, if: :devise_controller?
   before_action :set_locale
 
+  # before_filter :authenticate_user!, :set_current_user
```
Comment (davidwessman) [inline app/controllers/application_controller.rb]: Was this line in there before?
Prefer to remove lines than comment - we still have history in git 😄

File: db/migrate/20160223155800_create_audits.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class CreateAudits < ActiveRecord::Migration
+  def change
+    create_table :audits do |t|
+      t.references :auditable, polymorphic: true, index: true
+      t.references :user, index: true, foreign_key: true
+      t.references :updater, index: true
+      t.references :vote, index: true, foreign_key: true
+      t.json :audited_changes, default: {}, null: false
```
Comment (davidwessman) [inline db/migrate/20160223155800_create_audits.rb]: Perhaps this should be in it's own migration file - but I rather wipe the Heroku anyway 😄

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (navro) [inline app/controllers/application_controller.rb]: > Was this line in there before?
> Prefer to remove lines than comment - we still have history in git :smile:

That line was used in the example. I guess I forgot to remove it :)

File: app/helpers/votes_helper.rb [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      values << split_hash(key, value, model)
+    end
+    safe_join(values)
+  end
+
+  def split_hash(key, value, model)
+    val = ''
+    case key
```
Comment (davidwessman) [inline app/helpers/votes_helper.rb]: Not quite sure what hound means here...

File: app/models/audit.rb (hunk 2 above)
Comment (navro) [inline app/models/audit.rb]: It's not there in the raw file :)

**Comment by davidwessman** [comment]
Perhaps remove the commented line, otherwise +1

**Pull request merged**